tests/test_config.py

```
@@ -398,6 +398,49 @@ def test_duplicate_names_rejected(self) -> None:
+        """Repo without @token does not include token key in result."""
+        result = parse_additional_repos("infra:https://github.com/org/infra")
+        assert "token" not in result[0]
+
+    def test_multiple_repos_mixed_tokens(self) -> None:
+        """Some repos with token, some without."""
+        result = parse_additional_repos(
+            "infra:https://github.com/org/infra@tok1,"
+            "product:https://github.com/org/product"
+        )
+        assert len(result) == 2
+        assert result[0]["token"] == "tok1"  # pragma: allowlist secret
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to: "token"

(S105)